### PR TITLE
feat: allow headers in `partition`

### PR DESCRIPTION
**Pull request by MthwRobinson**
### Summary

Allows for passing in headers to `partition` if a URL is passed.

### Testing

```python
from unstructured.partition.auto import partition

url = "https://www.understandingwar.org/sites/default/files/Russian%20Offensive%20Campaign%20Assessment%2C%20April%2011%2C%202023.pdf"
elements = partition(url=url, headers={"Accept": "application/pdf"}, strategy="fast")
```

### CODE REVIEW

File: unstructured/partition/auto.py [hunk 1]
```diff
@@ -31,6 +31,7 @@ def partition(
     strategy: str = "hi_res",
     encoding: str = "utf-8",
     paragraph_grouper: Optional[Callable[[str], str]] = None,
+    headers: Dict[str, str] = {},
```
Comment (qued) [inline unstructured/partition/auto.py:34]: Should probably add this to the parameters in the docstring.

File: unstructured/partition/auto.py [hunk 2]
```diff
@@ -62,7 +63,11 @@ def partition(
...
 
     if url is not None:
-        file, filetype = file_and_type_from_url(url=url, content_type=content_type)
+        file, filetype = file_and_type_from_url(
+            url=url,
+            content_type=content_type,
+            headers=headers,
+        )
     else:
         filetype = detect_filetype(
```
Comment (qued) [inline unstructured/partition/auto.py:72]: Should we emit a warning if we end up in the `else` block and `headers` is being ignored?

**Review by qued — APPROVED** [review]
LGTM other than some suggestions. The testing code returns a list of elements.

File: unstructured/partition/auto.py (hunk 2 above)
Comment (MthwRobinson) [reply unstructured/partition/auto.py:72]: Oh yeah good idea, will add that in.

**Pull request merged**